statePattern.py

```diff
@@ -0,0 +1,115 @@
+import abc
+import defs
+
+# State machine implemented with state pattern
+
+
+class Context:
+
+    def __init__(self):
+        self.state = Off()
+
+    def work(self, pool):
+        self.state.handle(self, pool)
+
+
+class State(metaclass=abc.ABCMeta):
+
+    @abc.abstractmethod
+    def get_name(self):
+        pass
+
+    @abc.abstractmethod
+    def handle(self, context, pool):
+        pass
+
+
+class On(State):
+
+    def get_name(self):
+        return defs.ON
+
+    def handle(self, context, pool):
+        pool.handle_valve()
```

Comment:
This method could also return for example a boolean (or throw) and the return value could be checked for error situations in the valve and possibly in the future act on the information (send push notification to the mobile app etc) :)

Reply:
Maybe later...